pkg/util/testingjobs/pod/wrappers.go

```diff
@@ -52,9 +52,10 @@ func MakePod(name, ns string) *PodWrapper {
 			RestartPolicy: corev1.RestartPolicyNever,
 			Containers: []corev1.Container{
 				{
-					Name:      "c",
-					Image:     "pause",
-					Resources: corev1.ResourceRequirements{Requests: corev1.ResourceList{}, Limits: corev1.ResourceList{}},
+					Name:            "c",
+					Image:           "pause",
+					Resources:       corev1.ResourceRequirements{Requests: corev1.ResourceList{}, Limits: corev1.ResourceList{}},
+					SecurityContext: testing.UseDefaultSecurityContext(),
```

Comment:
```suggestion
```

We should avoid to add securityContext to MakePod function.
Instead of this, please define the wrapper function against PodWrapper, then create a test case to verify the situation where you want to check.

Reply:
I don't really understand what you mean.

I can add a new MakePods for the e2e integration for pods.

That is really the only thing I need. I was thinking that I can reuse the same function and just add this to that.

I am not verifying podSecurityContext at all. Right now, when I run these e2e suite on Openshift, we will fail on pod integration because Openshift requires securityContext on pods.

ex: https://github.com/openshift/kubernetes-sigs-kueue/blob/release-0.11/test/e2e/singlecluster/pod_test.go#L83

Would you prefer something like this where I only add the function to generate a more secure pod in the pod_tests?

Reply:
WDYT @tenzen-y @mimowo?

Reply:
I also prefer not to set securityContext directly in  MakePod, because it impacts unit tests unnecessarily.

I think @tenzen-y 's suggestion is to introduce a new function `DefaultSecurityContext` which will add the security context onto the PodWrapper. So we can chain like this in e2e tests:
```golang
MakePod().DefaultSecurityContext().PodSets().....Obj().
```
I like it more than current proposal.

Now, one downside is that we need to consistently remember to call `DefaultSecurityContext()` in e2e tests. So, instead we could introduce a MakePod function in `test/util/e2e` which would be a helper just for e2e tests, undernearth calling `return podtesting.MakePod(...).DefaultSecurityContext()`.

I'm ok with both approaches. cc @tenzen-y

Reply:
Introducing `DefaultSecurityContext` and use that only in E2E sounds great.
That is what I wanted to expect in my previous comment.

Reply:
The downside is that we need to remember calling it in all places for e2e tests, unless we introduce a helper like
MakePod just for e2e testing.

Reply:
Let's use `DefaultSecurityContext` explicitly, first.